tests/test_gzip_request.js

```diff
@@ -36,6 +36,35 @@ it('should accept and decode gzip encoded application/json', done => {
   req.end()
 })
 
+it('should accept and decode gzip encoded application/json, when headers come node-fetch style as array', done => {
+  const message = {
+    my: 'contents',
+  }
+
+  nock('http://example.test')
+    .post('/')
+    .reply(function (url, actual) {
+      expect(actual).to.deep.equal(message)
+      done()
+      return [200]
+    })
+
+  const req = http.request({
+    hostname: 'example.test',
+    path: '/',
+    method: 'POST',
+    headers: {
+      'content-encoding': ['gzip'],
+      'content-type': ['application/json'],
+    },
+  })
```

Comment:
```suggestion
  const scope = nock('http://example.test')
    .post('/')
    .reply(200, message)

  const req = http.request({
    hostname: 'example.test',
    path: '/',
    method: 'POST',
    headers: {
      'content-encoding': ['gzip'],
      'content-type': ['application/json'],
    },
  })

  req.on('response', () => {
    scope.done()
    done()
  })
```

Making assertions in reply callbacks isn't a good practice. Our preference is to use the matching built-in then assert the request was intercepted.